### PR TITLE
Adding e2e nav tests

### PR DESCRIPTION
**Pull request by Roughnick**
## PR Readiness Checklist

* General
    * [x] Verified all changes are related to feature described above (no scope creep)
* Branch Stuff
    * [x] Checked target and base branch (should be feature branch merging INTO `main`)
    * [x] Merged latest `main` into feature branch (so it has latest code)
    * [x] Resolved any conflicts between feature branch and `main`
* Test Stuff
    * [x] Added unit tests to cover modified / additional code
* GitHub Stuff (on the right 👉)
    * [x] Updated labels
    * [x] Linked issue to which this PR relates
    * [x] Added assignee
    * [x] Added at least one (1) reviewer

### CODE REVIEW

File: src/Nav/Nav.tsx [hunk 1]
```diff
@@ -130,6 +130,7 @@ const Nav: FC = () => {
 					className={classes.title}
 					role="heading"
 					variant="h4"
+					component="h1"
```
Comment (Wolven531) [inline src/Nav/Nav.tsx:133]: We do not need to specify that an `<h1>` tag is used here

```suggestion
```

File: test/cypress/integration/ConfigPage.spec.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+describe('Config Page', () => {
+	beforeEach(() => {
+		cy.visit('/config')
+		cy.injectAxe()
+	})
+
+	it('should load page', () => {
+		cy.get('h1').should('contain', 'Config Page')
```
Comment (Wolven531) [inline test/cypress/integration/ConfigPage.spec.js:8]: I'm guessing this is what required the update to `Nav.tsx` - this can be updated to `h4` since the `variant` is specified as `h4` in the Nav source

```suggestion
		cy.get('h4').should('contain', 'Config Page')
```

File: test/cypress/integration/HomePage.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+describe('Home Page', () => {
+	beforeEach(() => {
+		cy.visit('')
+		cy.injectAxe()
+	})
+
+	it('should load page', () => {
+		cy.title().should('eq', "Anthony and Vinny's CRA App")
+		cy.get('h1').should('contain', 'Home')
```
Comment (Wolven531) [inline test/cypress/integration/HomePage.spec.js:9]: Same as comment in other spec file

```suggestion
		cy.get('h4').should('contain', 'Home')
```

File: test/cypress/integration/PageNotFound.spec.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+describe('Search User Page', () => {
+	beforeEach(() => {
+		cy.visit('/ThisWillNeverBeAPage')
+		cy.injectAxe()
+	})
+
+	it('should load page', () => {
+		cy.get('h1').should('contain', 'Page Not Found')
```
Comment (Wolven531) [inline test/cypress/integration/PageNotFound.spec.js:8]: Same comment as other spec file

```suggestion
		cy.get('h4').should('contain', 'Page Not Found')
```

File: test/cypress/integration/SearchUserPage.spec.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+describe('Search User Page', () => {
+	beforeEach(() => {
+		cy.visit('/search-users')
+		cy.injectAxe()
+	})
+
+	it('should load page', () => {
+		cy.get('h1').should('contain', 'Search Users Page')
```
Comment (Wolven531) [inline test/cypress/integration/SearchUserPage.spec.js:8]: Same comment as other spec file

```suggestion
		cy.get('h4').should('contain', 'Search Users Page')
```

File: test/cypress/integration/SearchUserPage.spec.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+	})
+
+	describe('nav manu', () => {
```
Comment (Wolven531) [inline test/cypress/integration/SearchUserPage.spec.js:11]:
```suggestion
	describe('nav menu', () => {
```

File: test/cypress/integration/SearchUserPage.spec.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+describe('Search User Page', () => {
+	beforeEach(() => {
+		cy.visit('/search-users')
+		cy.injectAxe()
+	})
+
+	it('should load page', () => {
```
Comment (Wolven531) [inline test/cypress/integration/SearchUserPage.spec.js:7]: Should there be additional assertions here? perhaps on the search input and button?

File: test/cypress/integration/PageNotFound.spec.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+		cy.get('h5').should(
+			'contain',
+			'Page not found. How did you even get here?'
+		)
+	})
+
+	describe('nav manu', () => {
```
Comment (Wolven531) [inline test/cypress/integration/PageNotFound.spec.js:15]:
```suggestion
	describe('nav menu', () => {
```

File: test/cypress/integration/ConfigPage.spec.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+		cy.contains('button', 'Server Up?')
+		cy.contains('button', 'Token valid?')
+		cy.findByText('Riot Developer Site')
+			.should('have.attr', 'href')
+			.and('contain', 'https://developer.riotgames.com')
+	})
+
+	describe('nav manu', () => {
```
Comment (Wolven531) [inline test/cypress/integration/ConfigPage.spec.js:16]:
```suggestion
	describe('nav menu', () => {
```

File: tsconfig.json [hunk 10]
```diff
@@ -1,26 +1,20 @@
...
+		"forceConsistentCasingInFileNames": true,
+		"noFallthroughCasesInSwitch": true,
+		"module": "esnext",
+		"moduleResolution": "node",
+		"resolveJsonModule": true,
+		"isolatedModules": true,
+		"noEmit": true,
+		"jsx": "react-jsx"
+	},
+	"include": ["src", "test"]
```
Comment (Wolven531) [inline tsconfig.json:19]: I think this is fine but we need to confirm that updating this does not cause the `test` to be included in the final artifact (produced by `npm run build`)

**Review by Wolven531 — CHANGES_REQUESTED** [review]
Looks great so far! Requesting several changes 😀

File: src/Nav/Nav.tsx (hunk 1 above)
Comment (Roughnick) [reply src/Nav/Nav.tsx:133]: I re-added this cause I didn't catch it was removed from an earlier PR I had. It's an accessibility rule that is easy to cover. https://dequeuniversity.com/rules/axe/4.2/page-has-heading-one

File: tsconfig.json (hunk 10 above)
Comment (Roughnick) [reply tsconfig.json:19]: Who do I do that?

File: test/cypress/integration/SearchUserPage.spec.js (hunk 7 above)
Comment (Roughnick) [reply test/cypress/integration/SearchUserPage.spec.js:7]: I did attempt that but I ran in to fetch issues, something related to polyfill. So I created #49 to work on that later

File: src/Nav/Nav.tsx (hunk 1 above)
Comment (Roughnick) [reply src/Nav/Nav.tsx:133]: Also you can see I was originally testing for it but cypress was not ran when the line of code was removed, see the deleted page simplePage.spec.js

Comment (Wolven531) [reply src/Nav/Nav.tsx:133]: Hmmm - requiring a page to have a level one heading seem arbitrary, but I suppose if it's for accessibility, we can just style the h1 differently. Thanks for the explanation

File: test/cypress/integration/SearchUserPage.spec.js (hunk 7 above)
Comment (Wolven531) [reply test/cypress/integration/SearchUserPage.spec.js:7]: Gotcha - this is where you mentioned the other story to avoid scope creep. No problem, 😀

File: tsconfig.json (hunk 10 above)
Comment (Wolven531) [reply tsconfig.json:19]: 1. With `test` in the `include`, run `npm run build`
2. Navigate to the produced `dist` directory
3. Verify no `test` dir is in the `dist` directory - when app is deployed, `dist` is the directory (and its contents) that will be hosted online. If `test` code is up there, we are inflating our app size and shipping test code (both of which are not great, xD)

Comment (Wolven531) [reply tsconfig.json:19]: I used the following command in PowerShell - 

`cls; npm i; npm run build`

`cls` - clear screen
`npm i` - install NodeJS deps
`npm run build` - produces directory ready for deployment

Looks like by default, our output dir is named `build` (not `dist`), but the rest of the checks still follow. That being said, it can be a bit mundane to look through the generated files, so I just searched for docs on the TSConfig file and found some info on it -

https://www.typescriptlang.org/tsconfig#include

Looks like that property simply configures how the compiler will look for files/sources and does not affect build output (in our case, create-react-app is what configures the build output).

TLDR; - this change is fine

**Review by Wolven531 — APPROVED** [review]
Lgtm, 👍🏻. Nice work
